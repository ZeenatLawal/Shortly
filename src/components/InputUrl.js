import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { shortenedUrl } from '../redux/shortUrlData/shortUrl';

const InputUrl = () => {
  const dispatch = useDispatch();
  const [url, setUrl] = useState('');
  const [validity, setCustomValidity] = useState('');

  const onUrlChange = (e) => setUrl(e.target.value);
  const onInvalidUrl = () => setCustomValidity('Please add a link');
  const onInputUrl = () => setCustomValidity('');

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(shortenedUrl(url));
    sessionStorage.setItem('original', JSON.stringify(url));
    e.target.reset();
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Shorten a link here..."
          onChange={onUrlChange}
          onInvalid={onInvalidUrl}
          onInput={onInputUrl}
          required
        />
        <button type="submit">Shorten it!</button>
      </form>
      <p>{validity}</p>
    </div>
  );
};

export default InputUrl;
