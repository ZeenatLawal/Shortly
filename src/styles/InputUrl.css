.mid-section {
  background: linear-gradient(to bottom, white 50%, var(--light-violet) 50%);
}

.input-section {
  background-color: var(--dark-violet);
  background-image: url('../images/bg-shorten-mobile.svg');
  background-repeat: no-repeat;
  background-position: top right;
  padding: 1rem;
}

.shorten {
  background-color: var(--cyan);
  padding: 0.5rem 2rem;
  border-radius: 0.2rem;
  border: none;
  color: white;
}
