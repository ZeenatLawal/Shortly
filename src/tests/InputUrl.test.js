import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/configureStore';
import InputUrl from '../components/InputUrl';

describe('InputUrl component', () => {
  test('Snapshot test', () => {
    const input = renderer.create(
      <Provider store={store}>
        <InputUrl />
      </Provider>,
    )
      .toJSON();
    expect(input).toMatchSnapshot();
  });

  test('render url input', () => {
    render(
      <Provider store={store}>
        <InputUrl />
      </Provider>,
    );

    const input = screen.getByTestId('url-input');
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'url');
  });
});
