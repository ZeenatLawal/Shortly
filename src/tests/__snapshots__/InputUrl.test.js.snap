// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputUrl component Snapshot test 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      data-testid="url-input"
      onChange={[Function]}
      onInput={[Function]}
      onInvalid={[Function]}
      placeholder="Shorten a link here..."
      required={true}
      type="url"
    />
    <button
      type="submit"
    >
      Shorten it!
    </button>
  </form>
  <p>
    
  </p>
</section>
`;
